neptune/new/internal/credentials.py

```diff
@@ -17,15 +17,16 @@
 import base64
 import json
 import os
+from functools import lru_cache
 from typing import Optional, Dict
 
 from neptune.new import envs
 from neptune.new import ANONYMOUS, ANONYMOUS_API_TOKEN
 from neptune.new.exceptions import NeptuneInvalidApiTokenException, NeptuneMissingApiTokenException
 
 
+@lru_cache(maxsize=None, typed=True)
```

Comment:
Why do we need cache for Credentials?

Reply:
It was the easiest way to achieve caching the `create_http_client_with_auth` etc. functions. I was trying to achieve it via overwriting the `__hash__` etc. but I think it's achieved for complex objects via `id(...)` or something and we must explicitly pass same instances of arguments to be cached.

Reply:
We should find some other way. It looks just fishy.

Reply:
Caching factory method in Credentials?

Reply:
my preferred idea would be to convert Credentials to frozen dataclass (or namedtuple, if we're not feeling that fancy :P) + move logic from `__init__` to a factory method

Reply:
Ok. Converted to frozen dataclass + factory method